client/src/components/categorical/category.js

```diff
@@ -185,6 +196,83 @@ class Category extends React.Component {
     return err;
   };
 
+  categoryNameErrorMessage = () => {
+    const { newCategoryText } = this.state;
+    const err = this.editedCategoryNameError();
+    if (err === false) return null;
+
+    let markup = null;
+
+    if (err === "empty_string") {
+      markup = (
+        <span
+          style={{
+            display: "block",
+            fontStyle: "italic",
+            fontSize: 12,
+            marginTop: 5,
+            color: Colors.ORANGE3
+          }}
+        >
+          {"Category name cannot be blank"}
+        </span>
+      );
+    } else if (err === "already_exists") {
+      markup = (
+        <span
+          style={{
+            display: "block",
+            fontStyle: "italic",
+            fontSize: 12,
+            marginTop: 5,
+            color: Colors.ORANGE3
+          }}
+        >
+          {`${newCategoryText} already exists`}
```

Comment:
error messages sometimes contain the entered text (`newCategoryText`), sometimes do not.   Given that this is an inline component, I'm not sure it adds much value to replicate it.  And space is a premium, which also argues for omitting the user text in favor of a more verbose error message.